lib/extensions/attachable/active_record/base.rb

```diff
@@ -27,6 +41,51 @@ def files=(files)
         attachment_references.build(file: file)
       end
     end
+
+    private
+
+    # Delete the attachment references which are not in the content.
+    def update_attachment_references
+      return if attachment_references.empty?
+
+      attachment_reference_ids = active_attachment_reference_ids
```

Comment:
the record might have some new attachments which do not have an id, I think `mark_for_destruction` is cleaner ? otherwise need to do the reverse way and delete the attachments from the array.